lib/psulib_blacklight/solr_manager.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+require 'psulib_blacklight/solr_config'
+require 'psulib_blacklight/exceptions'
+
+module PsulibBlacklight
+  class SolrManager
+    ALLOWED_TIME_TO_RESPOND = 30
+
+    attr_reader :config
+
+    def initialize(config = SolrConfig.new)
+      @config = config
+      raise 'Cannot find Solr' unless solr_up?
+
+      upload_config unless configset_exists?
+    end
+
+    def create_collection
+      resp = connection.get(SolrConfig::COLLECTION_PATH,
+                            action: 'CREATE',
+                            name: collection_name_with_version,
+                            numShards: config.num_shards,
+                            replicationFactor: config.replication_factor,
+                            maxShardsPerNode: config.max_shards_per_node,
+                            "collection.configName": config.configset_name)
+
+      check_resp(resp)
+    end
+
+    def create_alias
+      raise SolrCollectionsNotFoundError unless collections_with_prefix.any?
+
+      resp = connection.get(SolrConfig::COLLECTION_PATH,
+                            action: 'CREATEALIAS',
+                            name: config.alias_name,
+                            collections: last_incremented_collection)
+
+      check_resp(resp)
+    end
+
+    def modify_collection
+      raise SolrCollectionsNotFoundError unless collections_with_prefix.any?
+
+      resp = connection.get(SolrConfig::COLLECTION_PATH,
+                            action: 'MODIFYCOLLECTION',
+                            collection: last_incremented_collection,
+                            "collection.configName": config.configset_name)
+
+      check_resp(resp)
+    end
+
+    def last_incremented_collection
+      collections_with_prefix.last
+    end
+
+    private
+
+      def collections_with_prefix
+        collections.grep /#{config.collection_name}/
+      end
+
+      def collection_name_with_version
+        @collection_name_with_version ||= "#{config.collection_name}_v#{next_collection_version}"
+      end
+
+      def next_collection_version
+        return 1 if collections&.grep(/#{config.collection_name}/)&.empty?
+
+        collections.map { |version| version.scan(/\d/).first.to_i }.flatten.max + 1
```

Comment:
@cdmo is collections here is sth like this ["psu_v10"]? 
When I tried ` ["psu_v10"].map { |version| version.scan(/\d/).first.to_i }.flatten.max + 1` it gives me 2, am I checking it wrong?

Reply:
I will leave now and never come back 🤦 🤦 🤦 🤦 byeeee

Reply:
😛